### ISSUE TITLE
Review and backport EOS patch about OPENSSL_CONF

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-core#994]
https://github.com/EOSIO/eos/commit/42baafc5015b87cdf439db1fa36421773c17a81e

**Comment by pmconrad** [comment]
This rings a bell. Not sure if I've had problems with this myself or if I've seen it in one of the obscure compiler issues that have been reported here.

**Comment by jmjatlanta** [comment]
As for the changes in openssl_scope, I can't see any fix there. 

The rest of the changes just look to be log a message when you try to do something when you're on the wrong strand using the "bnet" protocol. I don't believe we use strands, although I would assume the same applies to threads. Whether we should be logging when certain method calls are called from the wrong thread is probably out of the scope of this issue.

In short, someone smarter than I should make these changes, and if you can make me smarter with what I missed, I'd appreciate it.

**Comment by pmconrad** [comment]
This is what I had in mind: https://github.com/bitshares/bitshares-fc/pull/42
Can you test if the EOS change resolves that problem?

**Comment by jmjatlanta** [comment]
@pmconrad 

I am unable to reproduce the issue reported in https://github.com/bitshares/bitshares-fc/pull/42 on my mac. My guess is that issue is more compiler flag related (see https://stackoverflow.com/questions/19469887/segmentation-fault-with-boostfilesystem). But it could simply be I'm not testing correctly.

I don't have much insight as to what https://github.com/EOSIO/eos/commit/42baafc5015b87cdf439db1fa36421773c17a81e fixes. Possibly related to https://github.com/EOSIO/eos/issues/3458

**Comment by pmconrad** [comment]
I think the interesting part, i. e. the change in crypto/openssl, delays the construction of the static configurationFilePath until init_openssl is called. I'd assume that in certain cases the initialization happened too early, possibly before some internal boost stuff had been initialized properly. Not sure though.

**Comment by pmconrad** [comment]
I see no evidence that we have a problem in this area. Note that the EOS commit message talks about a "potential" fix.
Also, https://github.com/bitshares/bitshares-fc/pull/42 was unrelated and fixed by https://github.com/bitshares/bitshares-fc/pull/73 .
Closing.
